cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionUtilsTest.java

```diff
@@ -127,6 +129,49 @@ public void testProcessAlterPartitionReassignmentResult() throws Exception {
     EasyMock.reset(result);
   }
 
+  @Test
+  public void testGetInterBrokerReplicaTasksToReexecute() {
+    Set<TopicPartition> partitionsInMovement = Set.of(P0, P1);
+
+    ExecutionProposal proposalForPartition0 = EasyMock.mock(ExecutionProposal.class);
+    ExecutionProposal proposalForPartition1 = EasyMock.mock(ExecutionProposal.class);
+    ExecutionProposal proposalForPartition2 = EasyMock.mock(ExecutionProposal.class);
+
+    EasyMock.expect(proposalForPartition0.topicPartition()).andReturn(P0).anyTimes();
+    EasyMock.expect(proposalForPartition1.topicPartition()).andReturn(P1).anyTimes();
+    EasyMock.expect(proposalForPartition2.topicPartition()).andReturn(P2).anyTimes();
+    EasyMock.replay(proposalForPartition0);
+    EasyMock.replay(proposalForPartition1);
+    EasyMock.replay(proposalForPartition2);
+
+    ExecutionTask executionTaskForPartition0 = EasyMock.mock(ExecutionTask.class);
+    ExecutionTask executionTaskForPartition1 = EasyMock.mock(ExecutionTask.class);
+    ExecutionTask executionTaskForPartition2 = EasyMock.mock(ExecutionTask.class);
+
+    EasyMock.expect(executionTaskForPartition0.proposal()).andReturn(proposalForPartition0).anyTimes();
+    EasyMock.expect(executionTaskForPartition1.proposal()).andReturn(proposalForPartition1).anyTimes();
+    EasyMock.expect(executionTaskForPartition2.proposal()).andReturn(proposalForPartition2).anyTimes();
+    EasyMock.replay(executionTaskForPartition0);
+    EasyMock.replay(executionTaskForPartition1);
+    EasyMock.replay(executionTaskForPartition2);
+
+    // Case 1: all partitions in candidate tasks are already in movement. Expect nothing to re-execute
+    List<ExecutionTask> tasks1 = List.of(executionTaskForPartition0, executionTaskForPartition1);
+    List<ExecutionTask> tasksToReexecute1 = ExecutionUtils.getInterBrokerReplicaTasksToReexecute(partitionsInMovement, tasks1);
+    Assert.assertTrue(tasksToReexecute1.isEmpty());
+
+    // Case 2: some of partitions in candidate tasks are in movement and some are not. Expect some tasks to re-execute
+    List<ExecutionTask> tasks2 = List.of(executionTaskForPartition0, executionTaskForPartition1, executionTaskForPartition2);
+    List<ExecutionTask> tasksToReexecute2 = ExecutionUtils.getInterBrokerReplicaTasksToReexecute(partitionsInMovement, tasks2);
+    Assert.assertEquals(1, tasksToReexecute2.size());
+    Assert.assertEquals(P2, tasksToReexecute2.get(0).proposal().topicPartition());
+
+    // Case 3: Partitions of candidate tasks is subset of partitions in movement. Expect nothing to re-execute
+    List<ExecutionTask> tasks3 = List.of(executionTaskForPartition0);
+    List<ExecutionTask> tasksToReexecute3 = ExecutionUtils.getInterBrokerReplicaTasksToReexecute(partitionsInMovement, tasks3);
+    Assert.assertTrue(tasksToReexecute3.isEmpty());
+  }
```

Comment:
[Given that we may drop all mocks, this would irrelevant] When using mocks, it would be useful to `EasyMock.verify` them to make sure that all expected calls are made as many as the expected number of times.

Reply:
Sometimes if the returning values pass the assertion, it should imply the function is called for a certain times. In those cases, do we still need those verify?